PlayFabClientSDK/source/PlayFabHttp/PlayFabHttp.cs

```diff
@@ -41,28 +41,26 @@ public class PlayFabJsonSuccess<TResult> where TResult : PlayFabResultCommon
 
     public static class PlayFabHttp
     {
-        private static readonly IPlayFabHttp _http;
-
-        static PlayFabHttp()
+        public static async Task<object> DoPost(string urlPath, PlayFabRequestCommon request, string authType, string authKey, Dictionary<string, string> extraHeaders)
         {
-            var httpInterfaceType = typeof(IPlayFabHttp);
-            var types = typeof(PlayFabHttp).GetAssembly().GetTypes();
-            foreach (var eachType in types)
+            if (PlayFabSettings.TitleId == null)
+                throw new Exception("You must set your titleId before making an api call");
+            var transport = (ITransportPlugin)PluginManager.GetPlugin(PluginContract.PlayFab_Transport);
```

Comment:
Is it possible to make GetPlugin(...) templated such that we don't need to cast everytime we invoke it?

Reply:
I could make it templated but it won't save customers much typing as you'd still have to specify the type. I.e.:

(ITransportPlugin)PluginManager.GetPlugin(...)
vs
PluginManager.GetPlugin<ITransportPlugin>(...)


---
In reply to: [201478452](https://github.com/PlayFab/CSharpSDK/pull/97#discussion_r201478452) [](ancestors = 201478452)

Reply:
I was thinking something more along the lines of PluginManager.GetPlugin<ITransportPlugin>(...). In general I think requiring the customer to cast is less preferable than specifying the type via <>.

Reply:
Same amount of typing, but more code bloat (each specialization of a generic adds up a new type on runtime)

---
In reply to: [201491537](https://github.com/PlayFab/CSharpSDK/pull/97#discussion_r201491537) [](ancestors = 201491537)